x-pack/plugins/fleet/server/services/fleet_proxies.ts

```diff
@@ -79,14 +90,25 @@ export async function deleteFleetProxy(
   id: string,
   options?: { fromPreconfiguration?: boolean }
 ) {
-  const fleetServerHost = await getFleetProxy(soClient, id);
+  const fleetProxy = await getFleetProxy(soClient, id);
 
-  if (fleetServerHost.is_preconfigured && !options?.fromPreconfiguration) {
+  if (fleetProxy.is_preconfigured && !options?.fromPreconfiguration) {
     throw new FleetProxyUnauthorizedError(`Cannot delete ${id} preconfigured proxy`);
   }
+  const { outputs, fleetServerHosts } = await getFleetProxyRelatedSavedObjects(soClient, id);
+
+  if (
+    [...fleetServerHosts, ...outputs].some(
+      (fleetServerHostOrOutput) => fleetServerHostOrOutput.is_preconfigured
+    ) &&
+    !options?.fromPreconfiguration
+  ) {
+    throw new FleetProxyUnauthorizedError(
```

Comment:
nit: unit test coverage could be improved on this logic

Reply:
Good call, I just added one in https://github.com/elastic/kibana/pull/151205/commits/aae7819505c93ba3fcdd53716695c3f1a4564cd2